gtsfm/loader/astronet_loader.py

```diff
@@ -0,0 +1,231 @@
+
+from gtsam import Cal3Bundler, Pose3, Rot3, Point3, SfmTrack
+
+import gtsfm.utils.io as io_utils
+import gtsfm.utils.logger as logger_utils
+from gtsfm.common.image import Image
+from gtsfm.loader.loader_base import LoaderBase
+
+from thirdparty.colmap.scripts.python.read_write_model import read_model
+from thirdparty.colmap.scripts.python.read_write_model import Camera as ColmapCamera
+from thirdparty.colmap.scripts.python.read_write_model import Image as ColmapImage
+from thirdparty.colmap.scripts.python.read_write_model import Point3D as ColmapPoint3D
+
+
+logger = logger_utils.get_logger()
+
+
+class AstroNetLoader(LoaderBase):
+    """Simple loader class that reads an AstroNet data segment.
+
+    Refs:
+    - https://github.com/travisdriver/astronet
+    """
+
+    def __init__(
+        self,
+        data_dir: str,
+        use_gt_extrinsics: bool = True,
+        use_gt_sfmtracks: bool = False,
+        max_frame_lookahead: int = 2,
+    ) -> None:
+        """Initialize loader from a specified segment directory (data_dir) on disk.
+
+        <data_dir>/
+             ├── images/: undistorted grayscale images
+             ├── cameras.bin: camera calibrations (see https://colmap.github.io/format.html#cameras-txt)
+             ├── images.bin: 3D poses and 2D tracks (see https://colmap.github.io/format.html#images-txt)
+             └── points3D.bin: 3D tracks (see https://colmap.github.io/format.html#points3d-txt)
+
+
+        Args:
+            data_dir: path to directory containing the COLMAP-formatted data: cameras.bin, images.bin, and points3D.bin
+            use_gt_extrinsics (optional): whether to use ground truth extrinsics. Used only for comparison with
+                reconstructed values.
+            use_gt_sfmtracks (optional): whether to use ground truth tracks. Used only for comparison with reconstructed
+                values.
+            max_frame_lookahead (optional): maximum number of consecutive frames to consider for
+                matching/co-visibility. Any value of max_frame_lookahead less than the size of
+                the dataset assumes data is sequentially captured.
+
+        Raises:
+            FileNotFoundError if `data_dir` doesn't exist or image path does not exist.
+            RuntimeError if ground truth camera calibrations not provided.
+        """
+        self._use_gt_extrinsics = use_gt_extrinsics
+        self._use_gt_sfmtracks = use_gt_sfmtracks
+        self._max_frame_lookahead = max_frame_lookahead
+
+        # Use COLMAP model reader to load data and convert to GTSfM format.
+        if Path(data_dir).exists():
+            cameras, images, points3d = read_model(path=data_dir, ext=".bin")
+            self._calibrations, self._wTi_list, img_fnames, self._sfmtracks = self.colmap2gtsfm(
+                cameras, images, points3d, load_sfmtracks=use_gt_sfmtracks
+            )
+        else:
+            raise FileNotFoundError("No data found at %s." % data_dir)
```

Comment:
prefer to raise error immediately, and not nest (from [contributing](https://github.com/borglab/gtsfm/blob/master/CONTRIBUTING.md#python-style))

So I would prefer:
```python
if not Path(data_dir).exists():
    raise FileNotFoundError("No data found at %s." % data_dir)

cameras, images, points3d = read_model(path=data_dir, ext=".bin")
self._calibrations, self._wTi_list, img_fnames, self._sfmtracks = self.colmap2gtsfm(
```

Reply:
agreed. fixed